### ISSUE TITLE
Cast `0` time result incompatible with MySQL  by go-sql-driver

### ISSUE THREAD

**Issue by aytrack** [pingcap/tidb#29245]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```
package main

import (
	"database/sql"
	"fmt"
	"time"

	_ "github.com/go-sql-driver/mysql"
)

func main() {
	dsn := "root@tcp(127.0.0.1:4000)/test?parseTime=true"
	db, err := sql.Open("mysql", dsn)
	if err != nil {
		panic(err)
	}

	var zero []uint8
	row, err := db.Query("select cast(? as time);", time.Time{})
	if err != nil {
		panic(err)
	}

	for row.Next() {
		err = row.Scan(&zero)
		if err != nil {
			panic(err)
		}
	}
	fmt.Printf("string(zero) == \"\" is %+v\n", string(zero) == "")
}

```

### 2. What did you expect to see? (Required)
`string(zero)` should be `00:00:00`
### 3. What did you see instead (Required)
`string(zero)` is `""`
### 4. What is your TiDB version? (Required)
```sql
tidb_version() | Release Version: v5.3.0-alpha-1226-g21a4a7e44
Edition: Community
Git Commit Hash: 21a4a7e441fcd3da219a0b2480d32840f16063c2
Git Branch: master
UTC Build Time: 2021-10-27 10:01:46
GoVersion: go1.17.2
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false

```

**Comment by hawkingrei** [comment]
this SQL can reproduce this problem.

```
select cast("0000-00-00" as time)
```

**Comment by Defined2014** [comment]
Already fixed on v8.4.0, close this issue.
